tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1338,38 +1357,385 @@ static void test_rx_priv_interleave(void)
 	}
 
 	expected_beacon = BEACON_TYPE_SECURE;
-	ASSERT_TRUE(wait_for_beacon(NULL, NULL));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, NULL, NULL));
 
 	expected_beacon = BEACON_TYPE_PRIVATE;
-	ASSERT_TRUE(wait_for_beacon(private_beacon_check, false));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, private_beacon_check,
+				    &same_random));
 
 	/* IVU was started here */
-	ASSERT_TRUE(wait_for_beacon(private_beacon_check, false));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, private_beacon_check,
+				    &same_random));
 	ASSERT_EQUAL(0x02, beacon.flags);
 	ASSERT_EQUAL(0x0001, beacon.iv_index);
 
 	memset(&beacon, 0, sizeof(beacon));
 	expected_beacon = BEACON_TYPE_SECURE;
 
-	ASSERT_TRUE(wait_for_beacon(NULL, NULL));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, NULL, NULL));
 	ASSERT_EQUAL(0x02, beacon.flags);
 	ASSERT_EQUAL(0x0001, beacon.iv_index);
 
 	/* KR was started here */
-	ASSERT_TRUE(wait_for_beacon(NULL, NULL));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, NULL, NULL));
 	ASSERT_EQUAL(0x03, beacon.flags);
 	ASSERT_EQUAL(0x0001, beacon.iv_index);
 
 	expected_beacon = BEACON_TYPE_PRIVATE;
 
 	err = bt_mesh_private_beacon_key(net_key_new, &priv_beacon_key);
 
-	ASSERT_TRUE(wait_for_beacon(private_beacon_check, false));
+	ASSERT_TRUE(wait_for_beacon(beacon_scan_cb, BEACON_INTERVAL + 1, private_beacon_check,
+				    &same_random));
 	ASSERT_EQUAL(0x03, beacon.flags);
 	ASSERT_EQUAL(0x0001, beacon.iv_index);
 
 	PASS();
 }
+
+#if CONFIG_BT_MESH_GATT_PROXY
+
+#define BEACON_TYPE_PRIVATE_NET_ID  2
+#define BEACON_TYPE_PRIVATE_NODE_ID 3
+#define BEACON_TYPE_PRIVATE_LEN     28
+#define TEST_NET_IDX1               0
+#define TEST_NET_IDX2               1
+#define MAX_TIMEOUT ((CONFIG_BT_MESH_NODE_ID_TIMEOUT * 1000) / 6)
+
+struct pp_netkey_ctx {
+	uint8_t *net_key;
+	uint8_t net_id[8];
+	struct bt_mesh_key id_key;
+};
+
+static struct pp_netkey_ctx pp_net0 = {.net_key = (uint8_t *)test_net_key};
+static struct pp_netkey_ctx pp_net1 = {.net_key = (uint8_t *)test_net_key_secondary};
+
+struct priv_test_ctx {
+	uint8_t beacon_type;
+	uint16_t *node_id_addr;
+};
+
+static void pp_netkey_ctx_init(struct pp_netkey_ctx *net)
+{
+	int err;
+
+	err = bt_mesh_identity_key(net->net_key, &net->id_key);
```

Comment:
Please use `ASSERT_OK_MSG` here and below
Please revise all changes. There are many places for this macro. I do not want to create storm from similar messages.